neuraxle/distributed/streaming.py

```diff
@@ -0,0 +1,511 @@
+        Start multiple processes or threads with the worker function as a target.
+
+        :param context: execution context
+        :type context: ExecutionContext
+        :return:
+        """
+        target_function = worker_function
+        # if self.use_savers:
+        #    self.wrapped.save(context, full_dump=True)
+        #    target_function = with_step_loading(worker_function)
+
+        self.workers = []
+        for _, worker_arguments in zip(range(self.n_workers), self.additional_worker_arguments):
+            if self.use_threading:
+                p = Thread(target=target_function, args=(self, context, worker_arguments))
+            else:
+                p = Process(target=target_function, args=(self, context, worker_arguments))
+
+            p.daemon = True
+            p.start()
+            self.workers.append(p)
+
+    def stop(self):
+        """
+        Stop all of the workers.
+
+        :return:
+        """
+        if not self.use_threading:
+            [w.terminate() for w in self.workers]
+
+        self.workers = []
+        self.observers = []
+
+
+# def with_step_loading(wrapped_function):
+#    def wrapped_worker_function(step: QueueWorker, batches_to_process: Queue, context: ExecutionContext, additional_worker_arguments):
+#        step.set_step(FullDumpLoader(step.wrapped.name).load(context))
+#        wrapped_function(step, batches_to_process, context, additional_worker_arguments)
+
+#    return wrapped_worker_function
+
+
+def worker_function(step: QueueWorker, context: ExecutionContext, additional_worker_arguments):
+    """
+    Worker function that transforms the items inside the queue of items to process.
+
+    :param step: step to transform
+    :param additional_worker_arguments: any additional arguments that need to be passed to the workers
+    :param context: execution context
+    :type context: ExecutionContext
+    :return:
+    """
+    additional_worker_arguments = tuple(
+        additional_worker_arguments[i: i + 2] for i in range(0, len(additional_worker_arguments), 2)
+    )
+
+    for argument_name, argument_value in additional_worker_arguments:
+        step.get_step().__dict__.update({argument_name: argument_value})
+
+    while True:
+        task: QueuedPipelineTask = step.queue.get()
+        summary_id = task.data_container.summary_id
+        data_container = step.handle_transform(task.data_container, context)
+        data_container = data_container.set_summary_id(summary_id)
+        step.notify(QueuedPipelineTask(step_name=step.name, data_container=data_container))
+
+
+# (step_name, n_workers, step)
+# (step_name, n_workers, max_size, step)
+# (step_name, step)
+QueuedPipelineStepsTuple = Union[
+    Tuple[str, int, BaseStep],
+    Tuple[str, int, int, BaseStep],
+    Tuple[str, int, List[Tuple], BaseStep],
+    Tuple[str, int, List[Tuple], BaseStep],
+    Tuple[str, BaseStep]
+]
+
+
+class BaseQueuedPipeline(NonFittableMixin, CustomPipelineMixin, Pipeline):
```

Comment:
- remove NonFittableMixin. 
- fit transform sequentially
- transform in parallel 
- or fit transform in parallel if all steps are not fittable.

Reply:
@guillaume-chevalier this is more complicated then we thought.... to make this clean, we need to somewhat reuse MiniBatchSequentialPipeline logic to transform sequentially. I need to make the QueuedPipeline work with the Barrier() step like in MiniBatchSequentialPipeline...

Reply:
@guillaume-chevalier maybe the QueueJoiner should be Barrier that we dynamically inject in the constructor... !

Reply:
everything should be a BaseStep always has. always be.

Reply:
also, the cool thing is that a QueueJoiner is kind of a multiprocessing Barrier if you think about it ! wow everything comes together.